lib/workflows/send_errored_vm_email.py

```diff
@@ -0,0 +1,180 @@
+from typing import List, Optional, Union
+
+from email_api.emailer import Emailer
+from openstack_query import ServerQuery, UserQuery
+
+from enums.cloud_domains import CloudDomains
+from enums.query.props import ServerProperties
+from enums.query.query_presets import QueryPresetsGeneric, QueryPresetsDateTime
+
+from structs.email.email_params import EmailParams
+from structs.email.email_template_details import EmailTemplateDetails
+from structs.email.smtp_account import SMTPAccount
+
+
+def find_servers_with_errored_vms(
+    cloud_account: str, time_variable:int, from_projects: Optional[List[str]] = None
+):
+    """
+    :param cloud_account: string represents cloud account to use
+    :param from_projects: A list of project identifiers to limit search in
+    """
+
+
+    server_query = ServerQuery()
+    if time_variable > 0:
+        server_query.where(
+            QueryPresetsDateTime.OLDER_THAN, ServerProperties.SERVER_LAST_UPDATED_DATE, days=time_variable
+        )
+    server_query.where(
+        QueryPresetsGeneric.ANY_IN, ServerProperties.SERVER_STATUS, values=["ERROR"]
+    )
+    server_query.run(
+        cloud_account,
+        as_admin=True,
+        from_projects=from_projects if from_projects else None,
+        all_projects=not from_projects,
+    )
+
+    server_query.select("id", "name", "addresses")
+
+    if not server_query.to_props():
+        raise RuntimeError(
+            f"No servers found in [ERROR] state on projects "
+            f"{','.join(from_projects) if from_projects else '[all projects]'}"
+        )
+
+    server_query.append_from("PROJECT_QUERY", cloud_account, "name")
+    server_query.group_by("user_id")
+
+    return server_query
+
+
+def print_email_params(
+    email_addr: str, user_name: str, as_html: bool, error_table: str
+):
+    """
+    Print email params instead of sending the email
+    :param email_addr: email address to send to
+    :param user_name: name of user in openstack
+    :param as_html: A boolean which if selected will send an email, otherwise prints email details only
```

Comment:
Considering this function only prints emails surely even if `as_html` is true an email still wont be sent? With regards to this specific function

Reply:
Function is for testing, e.g. printing the output of an email so you can check it. Can probably be removed if no longer needed in production.